api/users/models.py

```diff
@@ -158,6 +168,24 @@ def get_full_name(self):
             return None
         return " ".join([self.first_name, self.last_name]).strip()
 
+    @property
+    def can_send_password_reset_email(self) -> bool:
+        with suppress(TypeError):
+            if self.last_password_reset_email_at < timezone.now() - timezone.timedelta(
+                seconds=settings.PASSWORD_RESET_EMAIL_COOLDOWN
+            ):
+                return True
+            return (
+                self.num_of_password_reset_emails_sent
+                < settings.MAX_PASSWORD_RESET_EMAILS
+            )
+
+        return True
```

Comment:
In response to the main thread on the PR, I tend to agree that we should do slightly more to prevent it but I do think that if we use a separate model this logic will become slightly cleaner / more flexible? 

Something like: 

```python
class UserPasswordResetRequest(models.Model):
    user = models.ForeignKey("users.FFAdminUser", related_name="password_reset_requests")
    requested_at = models.DateTimeField(auto_now_add=True)
```

Then the code above becomes something like: 

```python
def can_send_password_reset_email(self) -> bool:
    limit = timezone.now() - timezone.timedelta(
        seconds=settings.PASSWORD_RESET_EMAIL_COOLDOWN
    )
    return (
        self.password_reset_requests.filter(requested_at__gte=limit).count() 
        > settings.MAX_PASSWORD_RESET_EMAILS
    )
```

Then on a successful password reset, we just clear the table for that user.

(sidenote: as per my comment on another PR, I'd rather not query the database in properties if we can avoid it)

Reply:
Yeah, good point. Done